tests/dom/customElement.ts

```diff
@@ -1,5 +1,6 @@
 import userEvent from '#src'
-import {addListeners} from '#testHelpers'
+import {addListeners, setup} from '#testHelpers'
+import {setUISelection} from '#src/document'
 
 // It is unclear which part of our implementation is targeted with this test.
 // Can this be removed? Is it sufficient?
```

Comment:
I don't think this test suite should be extended.

Reply:
So your idea is to move these tests into copy, paste, cut test suites?

Reply:
The test in this file was introduced in https://github.com/testing-library/user-event/pull/320/files#diff-a1c186e56ef39b3e82d6781e7a934af052817d6ae42102d3c12368c9896e6e07

The thought there might have been to test for `.type()` what you're trying to test for `.copy()/.paste()/.cut()`, that the target for changes (the editable context) is correctly recognized to be inside the shadow tree.
Since then the implementation for `.type()` completely changed and so did the only assertion - an event screenshot.

I'd like to remove/replace this test, but therefore we need to understand which (if any) part of this test is important and which is just noise. If it's really just about recognizing the correct target, we should make sure to assert this on any API in the respective test suite. Maybe we should add a linting rule that ensures that `document.activeElement` is only used per the util that should document the edge cases - which of course we should test on it directly.

Anyway I would not add any new tests under that code comment, it just makes it harder to be confident when removing/replacing it.

Reply:
I am working on moving the tests at the moment. And I am throwing in a few goodies. :)

Reply:
So, I moved the tests and removed the test that was there. My reasoning is this: From what I see this is not a test of open shadow dom as the input element was directly passed to `user.type()`. I adapted the simple test I had to make sure that `type()` works directly on open shadow dom elements, although testing is going to be limited since `delegatesFocus` is not implemented in jsdom. I also think we should rather test recognizing the correct target, because the rest should be identical in open shadow dom.